### PR TITLE
NEXUS-7995: Bump HttpClient et al

### PR DESCRIPTION
**Pull request by cstamas**
Update HttpClient and HttpCore to:

HttpClient 4.3.6, release notes
HttpCore 4.3.3, release notes
Issue
https://issues.sonatype.org/browse/NEXUS-7995

CI
http://bamboo.s/browse/NX-OSSF527

### CODE REVIEW

File: buildsupport/httpclient/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
       <dependency>
         <groupId>org.apache.httpcomponents</groupId>
         <artifactId>httpclient</artifactId>
-        <version>4.3.5</version>
+        <version>4.3.6</version>
```
Comment (kellyrob99) [inline buildsupport/httpclient/pom.xml]: same property extraction question as on 2.11.x line

**Comment by kellyrob99** [comment]
+1 if CI is happy

**Comment by jdillon** [comment]
+1 if CI is happy

**Comment by cstamas** [comment]
http://bamboo.s/browse/NX-OSSF527-1 :+1:

**Pull request merged**